akka-actor/src/main/scala/akka/io/SimpleDnsCache.scala

```diff
@@ -112,3 +116,23 @@ object SimpleDnsCache {
     }
   }
 }
+
+sealed trait CachePolicy
+case object NeverCache extends CachePolicy
+case object CacheForever extends CachePolicy
+case class FiniteCache(ttl: FiniteDuration) extends CachePolicy {
+  require(ttl.toSeconds > 0)
+  import akka.util.JavaDurationConverters._
+  def getTtl: java.time.Duration = ttl.asJava
+}
+object FiniteCache {
+  // There's places where only a FiniteCache makes sense (DNS RFC says TTL is a positive 32 but integer)
+  // but we know the value can be cached effectively forever (e.g. the Lookup was the actual IP already)
+  val effectivelyForever: FiniteCache = FiniteCache(Int.MaxValue.seconds)
```

Comment:
Turns out I needed a `FiniteCache.effectivelyForever` which is typed `FiniteCache` but uses a value big enough.

The alternative was typing the `ttl` in all `DnsRecord` as `CachePolicy` and spread `asInstanceOf` in a few places (e.g. the `Ordering` type class).